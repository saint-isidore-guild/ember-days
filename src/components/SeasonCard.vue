<template>
  <v-col cols="3">
    <h2>{{ seasonName }}</h2>
    <h3>{{formattedDates}}</h3>
  </v-col>
</template>

<script>
import {calculateAllDates} from "@/utils/dateUtils";

export default {
  name: "SeasonCard",
  props: {startDate: Date, seasonName: String},
  computed: {
    formattedDates() {
      let allDates = calculateAllDates(this.startDate)
      const options = {month: 'long', day: 'numeric'}
      let dateStr = `${this.startDate.toLocaleDateString("en-US", options)}, `
      dateStr += `${this.startDate.getMonth() === allDates[1].getMonth() ? allDates[1].getDate() : allDates[1].toLocaleDateString("en-US", options)}, `
      dateStr += `${allDates[1].getMonth() === allDates[2].getMonth() ? allDates[2].getDate() : allDates[2].toLocaleDateString("en-US", options)}`
      return dateStr
    }
  }
}
</script>

<style scoped>

</style>