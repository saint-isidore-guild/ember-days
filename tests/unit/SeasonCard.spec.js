import { shallowMount } from '@vue/test-utils'
import SeasonCard from '@/components/SeasonCard'

describe('SeasonCard.vue', () => {
  it('renders props when passed', () => {
    const msg = 'Spring February 1, 3, 4'
    const wrapper = shallowMount(SeasonCard, {
      propsData: { seasonName: 'Spring', startDate: new Date(2020, 1, 1, 0, 0, 0) }
    })
    expect(wrapper.text()).toMatch(msg)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
