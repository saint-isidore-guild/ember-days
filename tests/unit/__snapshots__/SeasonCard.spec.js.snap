// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SesonCard.vue renders props when passed 1`] = `
<v-col-stub cols="12" md="3" tag="div">
  <h2>Spring</h2>
  <h3>February 1, 3, 4</h3>
</v-col-stub>
`;
